compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2697,10 +2698,47 @@ class Typer extends Namer
       case tree: Closure => cpy.Closure(tree)(tpt = TypeTree(pt)).withType(pt)
     }
 
+    /** Replace every top-level occurrence of a wildcard type argument by
```

Comment:
I'm not 100% clear on the notion of "top-level occurence" here, why is in an occurence in a branch of an AndOrType top-level, but not in the widened type of an `ExprType` for example ?

Reply:
It's because we have already widened the type before we pass it to `captureWildcards`.

Reply:
I'll make a note explaining that.